coderd/unhanger/detector_test.go

```diff
@@ -788,3 +822,17 @@ func TestDetectorMaxJobsPerRun(t *testing.T) {
 	detector.Close()
 	detector.Wait()
 }
+
+// wrapDBAuthz adds our Authorization/RBAC around the given database store, to
+// ensure the unhanger has the right permissions to do its work.
+func wrapDBAuthz(db database.Store, logger slog.Logger) database.Store {
+	accessControlStore := &atomic.Pointer[dbauthz.AccessControlStore]{}
+	var acs dbauthz.AccessControlStore = dbauthz.AGPLTemplateAccessControlStore{}
+	accessControlStore.Store(&acs)
+	return dbauthz.New(
+		db,
+		rbac.NewStrictCachingAuthorizer(prometheus.NewRegistry()),
+		logger,
+		accessControlStore,
+	)
+}
```

Comment:
There is a little `coderdtest` helper for the access control pointer.
```suggestion
// wrapDBAuthz adds our Authorization/RBAC around the given database store, to
// ensure the unhanger has the right permissions to do its work.
func wrapDBAuthz(db database.Store, logger slog.Logger) database.Store {
	return dbauthz.New(
		db,
		rbac.NewStrictCachingAuthorizer(prometheus.NewRegistry()),
		logger,
		coderdtest.AccessControlStorePointer(),
	)
}
```